website/source/docs/secrets/ad/index.html.md

```diff
@@ -0,0 +1,164 @@
+    `certificate`.
+
+3. Configure a role that maps a name in Vault to an account in Active Directory.
+When applications request passwords, password rotation settings will be managed by
+this role.
+
+    ```text
+    $ vault write ad/roles/my-application \
+        service_account_name="my-application@example.com"
+    ```
+    
+4. Grant "my-application" access to its creds at `ad/creds/my-application` using an 
+auth method like [AppRole](https://www.vaultproject.io/api/auth/approle/index.html).
+
+## Configuration
+
+The `config` endpoint configures the LDAP connection and binding parameters, as well as the password rotation configuration.
+
+### Password parameters
+
+* `ttl` (string, optional) - The default password time-to-live in seconds. Once the ttl has passed, a password will be rotated the next time it's requested.
+* `max_ttl` (string, optional) - The maximum password time-to-live in seconds. No role will be allowed to set a custom ttl greater than the `max_ttl`.
+* `password_length` (string, optional) - The desired password length. Defaults to 64. Minimum is 14. Note: to meet complexity requirements, all passwords begin with "?@09AZ".
+
+### Connection parameters
+
+* `url` (string, required) - The LDAP server to connect to. Examples: `ldap://ldap.myorg.com`, `ldaps://ldap.myorg.com:636`. This can also be a comma-delineated list of URLs, e.g. `ldap://ldap.myorg.com,ldaps://ldap.myorg.com:636`, in which case the servers will be tried in-order if there are errors during the connection process.
+* `starttls` (bool, optional) - If true, issues a `StartTLS` command after establishing an unencrypted connection.
+* `insecure_tls` - (bool, optional) - If true, skips LDAP server SSL certificate verification - insecure, use with caution!
+* `certificate` - (string, optional) - CA certificate to use when verifying LDAP server certificate, must be x509 PEM encoded.
+
+### Binding parameters
+
+* `binddn` (string, required) - Distinguished name of object to bind when performing user and group search. Example: `cn=vault,ou=Users,dc=example,dc=com`
+* `bindpass` (string, required) - Password to use along with `binddn` when performing user search.
+* `userdn` (string, optional) - Base DN under which to perform user search. Example: `ou=Users,dc=example,dc=com`
+* `upndomain` (string, optional) - userPrincipalDomain used to construct the UPN string for the authenticating user. The constructed UPN will appear as `[username]@UPNDomain`. Example: `example.com`, which will cause vault to bind as `username@example.com`.
+
+## FAQ
+
+### What if someone directly rotates an Active Directory password that Vault is managing?
+
+If an administrator at your company rotates a password that Vault is managing, the next time an application asks _Vault_ 
+for that password, Vault won't know it. 
+
+To maintain that application's up-time, Vault will need to return to a state of knowing the password. Vault will generate 
```

Comment:
Does this happen automatically, or is there something the operator does to force an update to get in sync?

Reply:
Automagically. :-)

Reply:
Added a little elaboration on the following line.